specs/ff-asn-selected-filter-cloud/ff-asn-selected-filter-cloud.md

```diff
@@ -1,86 +1,212 @@
 ```html
 <ff-asn-selected-filter-cloud blacklist="Price,Size">
-    <ff-asn-group-element>
-        <span slot="selected">{{name}}</span>
-        <span slot="unselected">{{name}}</span>
-    </ff-asn-group-element>
+    <div data-filter-container>
+         <span data-filter>{{name}}</span>
+    </div>
 </ff-asn-selected-filter-cloud>
 ```
 
 ### whitelist
-If you want to exclusively specify each filter group you can use the `whitelist` attribute.
+If you want to exclusively specify each filter **group**, you can use the `whitelist` attribute. Like in `blacklist`, you may specify as many groups as you need, separated by commas. The values must correspond to the group names returned by FACT-Finder.
 
 **Note**
-This attribute can not be used in conjunction with the **blacklist** attribute.
+This attribute cannot be used in conjunction with the `blacklist` attribute. Doing so will cause an exception to be thrown.
 
 #### example whitelist usage
 This example will whitelist only filters applied by the `Gender` and `Catgeory` groups.
 ```html
-<ff-asn-selected-filter-cloud whitelist="Gender,Category">
-    <ff-asn-group-element>
-        <span slot="selected">{{name}}</span>
-        <span slot="unselected">{{name}}</span>
-    </ff-asn-group-element>
-</ff-asn-selected-filter-cloud>
+<ff-filter-cloud whitelist="Gender,Category">
+    <div data-filter-container>
+        <span data-filter>{{name}}</span>
+    </div>
+</ff-filter-cloud>
 ```
 
+## Default templates
+
+### Setup
+```html
+<ff-filter-cloud></ff-filter-cloud>
+```
+
+### Rendered HTML
+```html
+<ff-filter-cloud>
+    <div data-filter-container>
+        <span data-filter>Filter 1</span>
+        <span data-filter>Filter 2</span>
+        <!-- ... -->
+        <span data-filter>Filter n</span>
+    </div>
+</ff-filter-cloud>
+```
+
+---
+
+### Setup
+```html
+<ff-filter-cloud>
+    <div data-filter-container></div>
+</ff-filter-cloud>
+```
+
+### Rendered HTML
+```html
+<ff-filter-cloud>
+    <div data-filter-container>
+        <span data-filter>Filter 1</span>
+        <span data-filter>Filter 2</span>
+        <!-- ... -->
+        <span data-filter>Filter n</span>
+    </div>
+</ff-filter-cloud>
+```
+
+---
+
+### Setup
+```html
+<ff-filter-cloud>
+    <div data-filter-container>
+        <span data-filter>
+            <span>Custom: {{name}}</span>
+            <span deselect>×</span>
+        </span>
+    </div>
+</ff-filter-cloud>
+```
+
+### Rendered HTML
+```html
+<ff-filter-cloud>
+    <div data-filter-container>
+        <span data-filter>
+            <span>Custom: Filter 1</span>
+            <span deselect>×</span>
+        </span>
+        <span data-filter>
+            <span>Custom: Filter 2</span>
+            <span deselect>×</span>
+        </span>
+        <!-- ... -->
+        <span data-filter>
+            <span>Custom: Filter n</span>
+            <span deselect>×</span>
+        </span>
+    </div>
+</ff-filter-cloud>
+```
+
+
+## Structural attributes
```

Comment:
I am not too happy with the term _Structural attributes_, as it is too broad for me. Maybe something like _Template specifier_ (these attributes are the hooks for our template engines, aren't they?)

Reply:
Dito, we currently call it just `properties` everywhere. I dont like to introduce a new naming just for one element

Reply:
See above:
> In the documentation we call it `directive`